src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,313 @@
+import os
+import logging
+import numpy as np
+from time import sleep
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+try:
+    import clr
+
+    lf_root = os.environ['LIGHTFIELD_ROOT']
+    automation_path = lf_root + '\PrincetonInstruments.LightField.AutomationV4.dll'
+    addin_path = lf_root + '\AddInViews\PrincetonInstruments.LightFieldViewV4.dll'
+    support_path = lf_root + '\PrincetonInstruments.LightFieldAddInSupportServices.dll'
+
+    addin_class = clr.AddReference(addin_path)
+    automation_class = clr.AddReference(automation_path)
+    support_class = clr.AddReference(support_path)
+
+    import PrincetonInstruments.LightField as lf
+
+    clr.AddReference("System.Collections")
+    clr.AddReference("System.IO")
+    from System.Collections.Generic import List
+    from System import String
+    from System.IO import FileAccess
+except Exception as e:
+    logger.error(f"Exception occurred during import: {e}")
+
+
+class LightfieldApp:
+    def __init__(self, visible):
+
+        self._addinbase = lf.AddIns.AddInBase()
+        self._automation = lf.Automation.Automation(visible, List[String]())
+        self._application = self._automation.LightFieldApplication
+        self._experiment = self._application.Experiment
+
+    @property
+    def automation(self):
+        return self._automation
+
+    @property
+    def addinbase(self):
+        return self._addinbase
+
```

Comment:
Just want to be clear, you are talking about just `addinbase` right? I'm asking this because if you remove property for `experiment` and `application` you wont be able to set anything :)

Reply:
So, the only class that uses `LightfieldApplicationManager` is the `SpectrometerConfig` class below, correct? Does it ever call these methods?

Reply:
I think this should work (also, I think AddInBase can be removed entirely from this class -- it doesn't seem to be used)

```
class LightfieldApplicationManager:
    def __init__(self, visible):
        self._automation = lf.Automation.Automation(visible, List[String]())
        self._application = self._automation.LightFieldApplication
        self._experiment = self._application.Experiment

    @property
    def experiment(self):
        return self._experiment

    def __del__(self):
        """
        Uses Python garbage collection method used to terminate
        AddInProcess.exe, if this is not done, LightField will not reopen
        properly.
        """
        self._automation.Dispose()
        logger.info('Closed AddInProcess.exe')

    def set(self, setting, value):
        ...

    def get(self, setting):
        ...

    def load_experiment(self, value):
        ...
        
    def save_current_experiment(self):
        ...
    
    def save_new_experient(self, value):
        ...

    def stop_scan(self):
        ...
    def acquire(self):
        # use self._application in this method -- note the underscore

    def close(self):
        """
        Closes the Lightfield application without saving the settings.
        """
        self._automation.Dispose()
        logger.info('Closed AddInProcess.exe')
```

We should keep the `experiment` property since it is accessed in the Spectrometer class